examples/DynamicCantileverCase/dynamic_cantilever.py

```diff
@@ -0,0 +1,145 @@
+import numpy as np
+from scipy.fft import fft, fftfreq
+from scipy.signal import find_peaks
+from elastica import *
+from dynamic_cantilever_helper import DynamicCantileverVibration
+
+
+def simulate_dynamic_cantilever_with(
+    density=2000,
+    n_elem=100,
+    final_time=300,
+    mode=0,
+    rendering_fps=30,  # For visualization
+):
+    class DynamicCantileverSimulator(BaseSystemCollection, Constraints, CallBacks):
+        pass
+
+    cantilever_sim = DynamicCantileverSimulator()
+
+    # Add test parameters
+    start = np.zeros((3,))
+    direction = np.array([1.0, 0.0, 0.0])
+    normal = np.array([0.0, 1.0, 0.0])
+    base_length = 1
+    base_radius = 0.02
+    base_area = np.pi * base_radius ** 2
+    youngs_modulus = 1e5
+
+    moment_of_inertia = np.pi / 4 * base_radius ** 4
+
+    dl = base_length / n_elem
+    dt = dl * 0.05
+    step_skips = int(1.0 / (rendering_fps * dt))
+
+    # Add Cosserat rod
+    cantilever = CosseratRod.straight_rod(
+        n_elem,
+        start,
+        direction,
+        normal,
+        base_length,
+        base_radius,
+        density,
+        0.0,
+        youngs_modulus,
+    )
+
+    # Add constraints
+    cantilever_sim.append(cantilever)
+    cantilever_sim.constrain(cantilever).using(
+        OneEndFixedBC, constrained_position_idx=(0,), constrained_director_idx=(0,)
+    )
+
+    end_velocity = 0.005
+    vibration = DynamicCantileverVibration(
```

Comment:
I don't like the `vibration` name here since it is not explicit. What do you think @bhosale2

Reply:
yeah, agree, since it's returning an analytical solution, maybe something like `analytical_cantilever_soln = AnalyticalDynamicCantilever(...)`?

Reply:
[0a9b1f6](https://github.com/GazzolaLab/PyElastica/pull/173/commits/0a9b1f614781bcfacf752ae3970a3b6a51dc197e)